File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] arr = new int[nums.length];
        
        int s=0 , e= nums.length-1;
        
        int lastNo= nums.length-1;
        
        while(lastNo>=0){
            if(Math.abs(nums[s])>Math.abs(nums[e])){
                arr[lastNo] = nums[s] * nums[s];
                s++;
            } 
            else{
                arr[lastNo] = nums[e] * nums[e];
                e--;
            }
            lastNo--;
        }
        
        return arr;
    }
}

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
       
        int start =0;
        int end = nums.size()-1;
        vector<int>ans;
     
        while(start<=end){
            int val1 = nums[start]*nums[start];
            int val2 = nums[end]*nums[end];
            
            if(val1>val2){
                 ans.push_back(val1);
                start++;
            }
            else{
               ans.push_back(val2);
                end--;
            }
        }
       reverse(ans.begin(),ans.end());
       
        return ans;
    }
};